tests/components/sensor/test_template.py

```diff
@@ -90,7 +86,7 @@ def test_invalid_name_does_not_create(self):
 
     def test_invalid_sensor_does_not_create(self):
         """Test invalid sensor."""
-        assert sensor.setup(self.hass, {
+        assert not bootstrap.setup_component(self.hass, 'sensor', {
```

Comment:
This will not work. The sensor component will always return `True`, even if some of it's platforms have failed to setup. 

Instead, look for states that have been setup. `assert self.hass.states.all() == []`